src/main/java/com/amazon/opendistroforelasticsearch/security/ssl/rest/OpenDistroSecuritySSLCertsInfoAction.java

```diff
@@ -0,0 +1,178 @@
+                final User user = (User)threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+                if(user == null || !adminDns.isAdmin(user)) {
+                    response = new BytesRestResponse(RestStatus.FORBIDDEN, builder);
+                } else {
+                    try {
+                        // Check if keystore initialised
+                        if (odsks != null) {
+                            builder.startObject();
+                            builder.field("http_certificates_list", generateCertDetailList(odsks.getHttpCerts()));
+                            builder.field("transport_certificates_list", generateCertDetailList(odsks.getTransportCerts()));
+                            builder.endObject();
+                            response = new BytesRestResponse(RestStatus.OK, builder);
+                        } else {
+                            builder.startObject();
+                            builder.field("message", "keystore is not initialized");
+                            builder.endObject();
+                            response = new BytesRestResponse(RestStatus.INTERNAL_SERVER_ERROR, builder);
+                        }
+                    } catch (final Exception e1) {
+                        log.error("Error handle request " + e1, e1);
+                        builder = channel.newBuilder();
+                        builder.startObject();
+                        builder.field("error", e1.toString());
+                        builder.endObject();
+                        response = new BytesRestResponse(RestStatus.INTERNAL_SERVER_ERROR, builder);
+                    } finally {
+                        if (builder != null) {
+                            builder.close();
+                        }
+                    }
+                }
+                channel.sendResponse(response);
+            }
+
+            /**
+             * Helper that construct list of certificate details.
+             * @param certs list of certificates.
+             * @return Array containing certificate details.
+             */
+            private List<Map<String, String>> generateCertDetailList(final X509Certificate[] certs) {
+                if (certs == null) {
+                    return null;
+                }
+                return Arrays.stream(certs)
+                    .map(cert -> {
+                        final String issuerDn = cert != null && cert.getIssuerX500Principal() != null ? cert.getIssuerX500Principal().getName(): "";
+                        final String subjectDn = cert != null && cert.getSubjectX500Principal() != null ? cert.getSubjectX500Principal().getName(): "";
+
+                        try {
+                            Collection<List<?>> sans = cert.getSubjectAlternativeNames();
+                        } catch (CertificateParsingException e) {
+                            e.printStackTrace();
+                        }
```

Comment:
This doesn't seem to do anything. Is this intended to just print an exception and continue ? Did you plan to stop executing the code that follows after this? Also, should it be print or logging.error(..) ?

Reply:
Was just supposed to log an error. Updated this.